platform/darwin/src/MGLShapeCollection.mm

```diff
@@ -13,6 +14,18 @@ + (instancetype)shapeCollectionWithShapes:(NSArray<MGLShape *> *)shapes {
 - (instancetype)initWithShapes:(NSArray<MGLShape *> *)shapes {
     if (self = [super init]) {
         _shapes = shapes.copy;
+        
+        for (MGLShape* shape in shapes ){
+            if ([shape conformsToProtocol:@protocol(MGLFeature)]) {
+                static dispatch_once_t onceToken;
+                dispatch_once(&onceToken, ^{
+                    NSLog(@"Shape collection will not retain attributes from objects conforming to MGLFeature protocol."
+                          @"Use MGLShapeCollectionFeature instead."
+                          @"This will be logged only once.");
+                });
+            }
+            break;
```

Comment:
This code breaks out of the loop, unconditionally, after considering the first shape in the array. I think the intention is to consider each of the shapes, breaking if one of them is a feature.